WordPress/gutenberg
# [RNMobile] memoize global styles in context to avoid unnecessary renders

## Description
In this PR I added memorization of globalStyle value that is passed through context to nested blocks. Unfortunately, the redux store works In an odd way - the innerBlocks in the parent block is always a new reference when we change the attribute of nested block. Because of that if we change something in the block that is nested in Column/Group etc the whole "inner blocks" are re-rendered and it also forces to generate the new `globalStyle`. I added use memo to the globalStyle, however, i couldn't simply use `attributes` and `wrapperProps.styles` as a dependency because they are a new reference. I had to use JSON.stringify which works fine but I do not like the idea ;(

This fixes the issue with a new reference of `globalStyle` that is passed to each block but I suggest also check why the redux store works like that and if it is necessary.

## How has this been tested?
To test if the app works exactly the same please follow these steps:
- Open post with some group blocks that have a background color, and paragraphs with text color set.
- Colors should be rendered properly


You can use this html for testing:
```

<div class="wp-block-group has-text-color has-background" style="background-color:#30c1ff;color:#fff586"><div class="wp-block-group__inner-container">
<p>1 level</p>


<div class="wp-block-group has-background" style="background-color:#24ad06"><div class="wp-block-group__inner-container">
<p class="has-text-color" style="color:#00628c">2 level</p>


<div class="wp-block-group has-text-color has-background" style="background-color:#f6f656;color:#139000"><div class="wp-block-group__inner-container">
<p>3 level</p>
</div></div>
</div></div>
</div></div>

```
To test if the fix actually prevents unnecessary renders, follow these steps: (It can be done by comparison of performance charts from the performance tab in the debugger but can be also tested by adding console logs)
In the performance tab:
- play with the editor, select paragraphs (in the same blog post as in previous tests)
- please compare charts with and without changes from this PR
By console logs.
- add console logs here https://github.com/WordPress/gutenberg/pull/28591/files#diff-5ccc3ba4a1df03d2c274a9932f16fe8ecd6adbe5ff2b3f8b380e85d60cd16ddbR53 and here https://github.com/WordPress/gutenberg/pull/28591/files#diff-5ccc3ba4a1df03d2c274a9932f16fe8ecd6adbe5ff2b3f8b380e85d60cd16ddbR55
- play with the editor, select paragraphs, change some text
- compare the count of console logs
- the console log inside useMemo should be called fewer times

## Screenshots 

## Types of changes
Bug-fix

## Checklist:
